app.py

```diff
@@ -58,6 +65,16 @@ def update_value_dict(self, value_list):
         value_dict['value_monthly'] = value_list
         return value_dict
 
+    def calculate_offset(self):
+        if self.historical_values['january'] == "undefined":
+            return self.percent_offset
+        else:
+            summed_historical = sum(self.historical_values.values())
+            summed_output = sum(self.energy_output)
+            percentage = (summed_output / summed_historical) * 100
```

Comment:
Lol, this is basically just figuring out how much energy the solar system (ha!) can provide the user.  So, say a user consumes 5000kWh of energy over the course of a year, and they design a system that will produce 4500kWh over the course of a year.  The solar system would potentially offset 90% of their energy usage (4500/5000) using solar energy instead of coal-fired energy, etc. over a year's time.